Source/WebCore/platform/network/NetworkStorageSession.cpp

```diff
@@ -494,12 +494,17 @@ std::optional<RegistrableDomain> NetworkStorageSession::findAdditionalLoginDomai
     return std::nullopt;
 }
 
-Vector<RegistrableDomain> NetworkStorageSession::storageAccessQuirkForTopFrameDomain(const TopFrameDomain& topDomain)
+Vector<RegistrableDomain> NetworkStorageSession::storageAccessQuirkForTopFrameDomain(const URL& topFrameURL)
 {
     for (auto&& quirk : updatableStorageAccessPromptQuirks()) {
-        auto& domainPairings = quirk.domainPairings;
-        auto entry = domainPairings.find(topDomain);
-        if (entry == domainPairings.end())
+        if (!quirk.triggerPages.isEmpty()) {
+            if (quirk.triggerPages.find(topFrameURL) == notFound)
+                continue;
+        }
```

Comment:
Nit - this might be a bit cleaner:

```suggestion
        if (quirk.triggerPages.contains(topFrameURL))
            continue;
```